src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -31,10 +31,13 @@ protected override void ExecuteCore()
             var destinationDirectory = Path.GetFullPath(AppHostDestinationDirectoryPath);
             ModifiedAppHostPath = Path.Combine(destinationDirectory, $"{appbaseName}{hostExtension}");
 
-            AppHost.Create(
-                AppHostSourcePath,
-                ModifiedAppHostPath,
-                AppBinaryName);
+            if (!File.Exists(ModifiedAppHostPath))
+            {
+                AppHost.Create(
+                    AppHostSourcePath,
+                    ModifiedAppHostPath,
+                    AppBinaryName);
```

Comment:
I forget now, is the content of AppHost a pure function of these inputs, and that's why we don't need to recreate it?

I realize this is how it was before, just in a different place. This is going to need to change when we start putting more stuff in apphost (version resources, icon, etc.)

Reply:
For EmbedAppNameInHost task, the old publish, it is true. only the dll name is embedded, but when dll name is changed, exe name is changed as well.

for generate shim(the only other user), I did use hash to make sure incremental build will be triggered when property changes